core/drivers/crypto/versal/ecc.c

```diff
@@ -99,13 +100,30 @@ static TEE_Result ecc_get_key_size(uint32_t curve, size_t *bytes, size_t *bits)
 	return TEE_SUCCESS;
 }
 
-static void crypto_bignum_bn2bin_eswap(uint32_t curve,
-				       struct bignum *from, uint8_t *to)
+static void eswap(size_t len, const uint8_t *from, uint8_t *to)
 {
-	uint8_t pad[66] = { 0 };
+	uint8_t *pad = calloc(1, len);
 	uint8_t tmp = 0;
 	size_t i = 0;
 	size_t j = 0;
+
+	if (!pad)
+		panic();
```

Comment:
It seems overkilling to panic here. Should better return a error code.

That said, you don't need a allocation here, simply loop on `to[i++] = from[j--]` or like.

Reply:
yeah I can do that, just need to remove the case where I used the the same buffer in from/to fields. let me do that.